sycl-fusion/passes/kernel-fusion/SYCLSpecConstMaterializer.cpp

```diff
@@ -294,18 +279,22 @@ bool SYCLSpecConstMaterializer::readMetadata() {
 
 PreservedAnalyses SYCLSpecConstMaterializer::run(Function &F,
                                                  FunctionAnalysisManager &) {
+  if (const char *DebugEnv = std::getenv("SYCL_MATERIALIZER_DEBUG"))
+    if (0 == strcmp(DebugEnv, DEBUG_TYPE))
+      llvm::setCurrentDebugType(DEBUG_TYPE);
```

Comment:
```suggestion
#ifndef NDEBUG
  if (const char *DebugEnv = std::getenv("SYCL_JIT_COMPILER_DEBUG"))
    llvm::setCurrentDebugType(DebugEnv);
#endif // NDEBUG
```

Calling this `SYCL_JIT_COMPILER_DEBUG` so we can reuse in other JIT compiler entrypoints. I'd move this to `materializeSpecConstants` (materialization entrypoint) just in case users want to debug other passes.

Reply:
Yeap, makes sense! Will change `strcmp` to `strstr` to do a sub-string search.

Updated the env variables docs to reflect the change as well.